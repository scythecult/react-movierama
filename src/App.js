import { useDispatch, useSelector } from "react-redux";
import { Route, Routes, Navigate } from "react-router-dom";

function App() {
  return <h1>jello</h1>;
}

export default App;
