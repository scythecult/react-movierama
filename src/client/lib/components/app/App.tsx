import { Container } from '../container/Container';
import { Info } from '../info/Info';
import { Legend } from '../legend/Legend';
import { MoviePicker } from '../movie-picker/MoviePicker';
import { Screen } from '../screen/Screen';
import { Seats } from '../seats/Seats';
import styles from './styles.module.css';

export const App = () => {
  return (
    <div className={styles.layout}>
      <MoviePicker />
      <Legend />
      <Container>
        <Screen />
        <Seats />
      </Container>
      <Info />
    </div>
  );
};
