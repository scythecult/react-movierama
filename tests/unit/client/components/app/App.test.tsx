import { render } from '@testing-library/react';
import { App } from '@/client/lib/components/app/App';

describe('App', () => {
  test('should correspond default layout', () => {
    const result = render(<App />);

    expect(result.container).toMatchSnapshot();
  });
});
