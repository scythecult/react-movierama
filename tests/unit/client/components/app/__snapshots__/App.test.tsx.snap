// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should correspond default layout 1`] = `
<div>
  <div
    class="_layout_38a165"
  >
    <div
      class="_moviePicker_2e4be8"
    >
      <label
        for="movie"
      >
        Pick a movie:
      </label>
      <select
        id="movie"
      >
        <option
          value="Avengers: Endgame"
        >
          Avengers: Endgame
           ($
          10
          )
        </option>
        <option
          value="Joker"
        >
          Joker
           ($
          12
          )
        </option>
        <option
          value="Toy Story 4"
        >
          Toy Story 4
           ($
          8
          )
        </option>
        <option
          value="The Lion King"
        >
          The Lion King
           ($
          9
          )
        </option>
        <option
          value="Matrix"
        >
          Matrix
           ($
          1
          )
        </option>
      </select>
    </div>
    <ul
      class="_legend_c88f57"
    >
      <li
        class="_legendItem_c88f57"
      >
        <button
          class="_seat_357f74 _disabled_357f74"
          type="button"
        />
        <small>
          N/A
        </small>
      </li>
      <li
        class="_legendItem_c88f57"
      >
        <button
          class="_seat_357f74 _selected_357f74 _disabled_357f74"
          type="button"
        />
        <small>
          Selected
        </small>
      </li>
      <li
        class="_legendItem_c88f57"
      >
        <button
          class="_seat_357f74 _occupied_357f74"
          disabled=""
          type="button"
        />
        <small>
          Occupied
        </small>
      </li>
    </ul>
    <div
      class="_container_72892e"
    >
      <div
        class="_screen_a4f98b"
      />
      <div
        class="_seats_fafe62"
      >
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74 _occupied_357f74"
          disabled=""
          type="button"
        />
        <button
          class="_seat_357f74 _occupied_357f74"
          disabled=""
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
      </div>
      <div
        class="_seats_fafe62"
      >
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
      </div>
      <div
        class="_seats_fafe62"
      >
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74 _occupied_357f74"
          disabled=""
          type="button"
        />
        <button
          class="_seat_357f74 _occupied_357f74"
          disabled=""
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
      </div>
      <div
        class="_seats_fafe62"
      >
        <button
          class="_seat_357f74 _occupied_357f74"
          disabled=""
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
      </div>
      <div
        class="_seats_fafe62"
      >
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74 _occupied_357f74"
          disabled=""
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
        <button
          class="_seat_357f74"
          type="button"
        />
      </div>
    </div>
    <p
      class="_info_7377e4"
    >
      You have selected 
      <span>
        0
      </span>
       seats for a price of
      <span>
         $
        0
      </span>
    </p>
  </div>
</div>
`;
